changefeedccl: add static labels to changefeed stage metrics

KeithCh opened the pull request:
This commit adds static labels to changefeed stage metrics.

Resolves: #156290

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/156901)
<!-- Reviewable:end -->

log-head reviewed (approved):
Looks good! Wondering if you think we should add a unit test for using these through the /metrics endpoint? I did cursory manual testing on this branch and the static labeled metrics show up.

asg0451 reviewed (changes requested):
as discussed, i dont want to lose 'changefeed.emitted.messages' just for this. if there isnt a good way to static labelify that one while maintaining the original then lets just not

KeithCh committed:
changefeedccl: add static labels to changefeed stage metrics

This commit adds static labels to changefeed stage metrics.

Resolves: #156290

Release note: None

KeithCh commented:
TFTRs!

bors r=asg0451,log-head
